tools/rosgraph/src/rosgraph/impl/graph.py

```diff
@@ -200,9 +203,9 @@ def __init__(self, start, end, label=''):
         self.start = start
         self.end = end
         self.label = label
-        self.key = "%s|%s"%(self.start, self.label)
+        self.key = (self.start, self.label)
```

Comment:
All these slots are public API of an edge and therefore can't change their value like this.